### DISCUSSION TITLE
Germany: Ostrom (2024)

### DISCUSSION THREAD

**Discussion by arboeh** [JaccoR/hass-entso-e#148] [Price Modifyer Templates]
Preiszusammensetzung laut AGB: https://www.ostrom.de/terms-and-conditions

- Kosten für den Bezug (inklusive Erzeugung) des Stroms
- Vertrieb
- Stromsteuer
- variable Netzentgelte
- Konzessionsabgaben
- Umlage nach Kraft-Wärme-Kopplungsgesetz (KWK-Umlage)
- Umlage nach § 19 Abs. 2 Strom NEV (§ 19 StromNEV-Umlage)
- Umlage nach § 17f. EnWG (Offshore-Netzumlage)
- Umlage nach § 18 der Verordnung zu abschaltbaren Lasten (Umlage für abschaltbare Lasten)
- Herkunftszertifikate
- Sonstiges

Ich habe mir das Ganze mal für Nordrhein-Westfalen zusammengesucht und -gebastelt, aber mir fehlen noch drei Werte "Vertrieb", "Herkunftszertifikate" und "Sonstiges":

- Kosten für den Bezug (inklusive Erzeugung) des Stroms: Aktueller Marktpreis
- Vertrieb: ?
- Stromsteuer: 0.0205 EUR/kWh
- variable Netzentgelte NRW: 0.065 EUR/kWh
- Konzessionsabgaben: ca. 0.01574 EUR/kWh
- Umlage nach Kraft-Wärme-Kopplungsgesetz (KWK-Umlage): 0.00275 EUR/kWh
- Umlage nach § 19 Abs. 2 Strom NEV (§ 19 StromNEV-Umlage): 0.00643 EUR/kWh
- Umlage nach § 17f. EnWG (Offshore-Netzumlage): 0.00656 EUR/kWh
- Umlage nach § 18 der Verordnung zu abschaltbaren Lasten (Umlage für abschaltbare Lasten): 0 EUR/kWh
- Herkunftszertifikate: ?
- Sonstiges: ?

Mein Template sieht zur Zeit so aus, VAT tariff ist 0.19:

```
{% set s = {
  "distribution": 0,
  "electricity_tax": 0.0205,
  "variable_network_fees": 0.065,
  "concession_fees": 0.01574,
  "kwk": 0.00275,
  "nev": 0.00643,
  "offshore": 0.00656,
  "ablav": 0,
  "guarantees_of_origin": 0,
  "misc": 0
}
%}
{{(current_price + s.distribution + s.electricity_tax + s.variable_network_fees + s.concession_fees + s.kwk + s.nev + s.offshore + s.ablav + s.guarantees_of_origin + s.misc) | float}}
```

Vielleicht bekommen wir das hier gemeinsam rund... :-)

**Comment by typxxi** [comment]
Ich frag mich, wann und wie Tibber das mal offenlegt, denn die buddeln alles in eine Summe bei uns. Das sind dann 20,2 Cent on top auf den Netto Börsenpreis und darin ist dann auch die Mehrwertsteuer enthalten für den Nettowert dieser Gebühren.
ich habe damit eber alles auf 0 und 

`"misc": 20.2`
so einfach und sagt am Ende nix mehr aus.

**Comment by typxxi** [comment]
Ich hatte bis Anfang der Woche nordpool als Datenlieferanten für Tibberpreise und habe das [Home Assistant Energiemanagement AIO](https://www.creatingsmarthome.com/index.php/2024/07/27/home-assistant-nord-pool-cheapest-hours-with-aio-energy-management/) eines Finnen genutzt, was unter HA preisabhängige Automationen erlaubt wie einschalten einer Last in den billigsten 2 h oder AC Laden des Haus Akku im Winter.

Da Nordpool aber die API umgestellt hat und nun Deutschland nicht mehr unter DE-LU sondern GER führt, sind dabei wohl falsche Preise ins Spiel gekommen, die man nur als unbrauchbar bezeichnen kann, weil die Spitzenpreise von gestern Abend bei Tibber bei unter 40 Cent lagen, während Nordpool plötzlich 47 Cent wollte. Und so passt kein Preis mehr zu Tibber. Tags zuvor lag die Lücke bei 27,8 Cent zu 31 Cent.
DAVOR waren es mit dem richtigen Template satte 0,01 Cent bei 1 von 6 Vergleichen, sonst immer nur 0,00 Cent !

**Da Nordpool als Quelle ausfällt und das AIO daneben nur noch ENTSO-e unterstützt, werde ich nun wieder mal migrieren.**

Die Ursache für o.g. Abweichungen von bis 7 Cent liegt auch nicht im Template, denn das hat sich nicht verändert und blieb im yaml Sensor setup weiterhin wie zuvor bei 

`{{ (0.2072 + (current_price * 0.19)) | float }}`

Dein hier dokumentiertes Model weicht allerdings ordentlich von dem der **"aktuellen Doku"**, weil Du keine VAT / Mehrwertsteuer mehr eingebunden hast - oder aber die Doku hier auf der 1. Seit ist alt mit der VAT im yaml code  wie hier zu sehen

![image](https://github.com/user-attachments/assets/71bb1f07-1ce9-4361-b05d-68aaf55ef8e6)

und das wurde wohl abgelöst durch UI Abfrage nach VAT tarif beim Einrichten hier
![image](https://github.com/user-attachments/assets/a2d6700d-3fff-46cb-bbaf-385bd5cd71b5)
 

Ich werden mal beginnen mit dem hier 
```
{% set s = {
  "distribution": 0.0,
  "electricity_tax": 0.0,
  "variable_network_fees": 0.0,
  "concession_fees": 0.0,
  "kwk": 0.0,
  "nev": 0.0,
  "offshore": 0.0,
  "ablav": 0.0,
  "guarantees_of_origin": 0.0,
  "misc": 0.2072
}
%}
{{(current_price + s.distribution + s.electricity_tax + s.variable_network_fees + s.concession_fees + s.kwk + s.nev + s.offshore + s.ablav + s.guarantees_of_origin + s.misc) | float}}
```

Mal sehen, wie das zu den Tibber Preisen passen wird und hoffentlich nicht wieder 7 Cent mehr anzeigen wird als das, was Tibber wirklich kostet.

**Comment by typxxi** [comment]
Das funktioniert so überhaupt nicht mit den Abgaben. 
26,6 Cent laut Tibber App und Tibber HA Integration / Sensor 
32,1 Cent laut entso-e sensor war 
**+5,5 Cent zu viel**

Also hab ich die 19% Mehrwertsteuer aus den Tibber "Netzentgelten" rausgezogen und 
"misc": geändert von **`0.2072`** durch 1.19 geteilt in **`0.174117647059`**

26,6 Cent laut Tibber App und Tibber HA Integration / Sensor 
28,1 Cent laut entso-e sensor 
**+1,5 Cent zu viel**

Besser, aber nicht wirklich gut.
Tibber Integration
![image](https://github.com/user-attachments/assets/cbf88776-cf34-4abc-8d48-15aa3f711c00)
Entso-E Integration
![image](https://github.com/user-attachments/assets/646a8254-4f19-4184-bc91-7f30e6e415f2)

Ich kann ja nun auch nicht mehr machen, aber ich vermute, dass da etwas mit der Mehrwertsteuer auf den Börsenpreis falsch läuft.

Bei Tibber 26,6 Cent und 20,72 Cent Netzentgelte (inkl. 19%) bliebe als Börsenpreis 5,88 Cent (inkl. 19%)
Würde ich aus dem Börsenpreis die 19% rausrechnen, dann wären das 0,96 Cent und netto 4,94

Selbst nach Abzug von 0,96 Cent bliebe noch eine Differenz von 0,54 Cent zu viel.

Letzter Stand des Modifyer Template
```
{% set s = {
  "distribution": 0.0,
  "electricity_tax": 0.0,
  "variable_network_fees": 0.0,
  "concession_fees": 0.0,
  "kwk": 0.0,
  "nev": 0.0,
  "offshore": 0.0,
  "ablav": 0.0,
  "guarantees_of_origin": 0.0,
  "misc": 0.174117647059
}
%}
{{(current_price + s.distribution + s.electricity_tax + s.variable_network_fees + s.concession_fees + s.kwk + s.nev + s.offshore + s.ablav + s.guarantees_of_origin + s.misc) | float}}
```

**Comment by typxxi** [comment]
Ich habe nordpool v0.0.15 nun zum Laufen gebracht.
0 Cent Abweichung zu Tibber, weil ich die Fehler darin gefunden habe, denn jemand hat fäschlicherweise 23% Mehrwertsteuer tief im Code für Deutschland hinterlegt, womit keiner in Deutschland rechnen würde.
